comms/comms.go

```diff
@@ -78,3 +83,14 @@ func NewToDDComms(cfg config.Config) (*toddComms, error) { // TODO: Return Packa
 	return &tc, nil
 
 }
+
+// NewAgentComms returns a comms instance configured for agent usages.
+//
+// TODO: accept an interface for cache instead of concrete type
+func NewAgentComms(cfg config.Config, ac *cache.AgentCache) (*toddComms, error) {
```

Comment:
Exactly, I was trying to limit the scope of the changes to the `agent/cache` package as much as possible.  At the time comms needed access to the agent cache, but only when being used by the agent, so I did the simple thing and added a special constructor just for the agent. The server and client still use `NewToDDComms`.

I didn't add a comment because I wasn't sure what was temporary and what wasn't. I did know that the `Package.setAgentCache()` method need to be removed eventually, so that did get a comment. As it turned out, when I got to the comms package in my `refactor-comms` branch, the agent cache dependency was decoupled from comms and I collapsed both constructors and renamed them to simply `New()`.